observability/observability.go

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observability
+
+import (
+	"context"
+
+	"google.golang.org/grpc/grpclog"
+)
+
+var (
+	logger = grpclog.Component("observability")
+)
+
+// Start is the opt-in API for gRPC Observability plugin.
+//   - it loads observability config from environment;
+//   - it registers default exporters if not disabled by the config;
+//   - it sets up binary logging sink against the logging exporter.
+func Start(ctx context.Context) {
+	config := parseObservabilityConfig(ctx)
+	// If the default logging exporter is not disabled, register one.
+	if config.ExporterConfig.ProjectId != "" && !config.ExporterConfig.DisableDefaultLoggingExporter {
+		createDefaultLoggingExporter(ctx, config.ExporterConfig.ProjectId)
+	}
+
+	startLogging(config)
+}
+
+// End is the clean-up API for gRPC Observability plugin. It is expected to be
+// invoked in the main function of the application. The suggested usage is
+// "defer observability.End()". This function also flushes data to upstream, and
+// cleanup resources.
+func End() {
```

Comment:
What happens if this is called while a grpc client or server is still open?

Reply:
The call is removed.

Once we closed exporter, the sink will be a no-op sink, so it's good enough.